cmd/ssh_key_delete.go

```diff
@@ -34,20 +32,27 @@ func (c *computeSSHKeyDeleteCmd) cmdPreRun(cmd *cobra.Command, args []string) er
 }
 
 func (c *computeSSHKeyDeleteCmd) cmdRun(_ *cobra.Command, _ []string) error {
-	ctx := exoapi.WithEndpoint(
-		gContext,
-		exoapi.NewReqEndpoint(account.CurrentAccount.Environment, account.CurrentAccount.DefaultZone),
-	)
+	ctx := gContext
 
 	if !c.Force {
 		if !askQuestion(fmt.Sprintf("Are you sure you want to delete SSH key %s?", c.Name)) {
 			return nil
 		}
 	}
 
-	var err error
-	decorateAsyncOperation(fmt.Sprintf("Deleting SSH key %s...", c.Name), func() {
-		err = globalstate.EgoscaleClient.DeleteSSHKey(ctx, account.CurrentAccount.DefaultZone, &egoscale.SSHKey{Name: &c.Name})
+	err := decorateAsyncOperations(fmt.Sprintf("Deleting SSH key %s...", c.Name), func() error {
+		op, err := globalstate.EgoscaleV3Client.DeleteSSHKey(ctx, c.Name)
+
+		if err != nil {
+			return err
+		}
+
+		_, err = globalstate.EgoscaleV3Client.Wait(ctx, op, v3.OperationStateSuccess)
+		if err != nil {
+			return err
+		}
+
+		return nil
```

Comment:
Since here you are not modifying/wrapping the error, you can simplify the code and return directly `decorateAsyncOperations` error

```Golang
	return decorateAsyncOperations(fmt.Sprintf("Deleting SSH key %s...", c.Name), func() error {
		op, err := globalstate.EgoscaleV3Client.DeleteSSHKey(ctx, c.Name)

		if err != nil {
			return err
		}

		_, err = globalstate.EgoscaleV3Client.Wait(ctx, op, v3.OperationStateSuccess)
		return err
	})
```

then we remove all those useless return nil, and error is returned directly, err variable will be nil if not error ;)